src/test/kotlin/no/nav/syfo/behandler/BehandlerServiceSpek.kt

```diff
@@ -0,0 +1,264 @@
+package no.nav.syfo.behandler
+
+import io.ktor.server.testing.*
+import io.mockk.clearAllMocks
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.behandler.database.getBehandlerDialogmeldingForPerson
+import no.nav.syfo.behandler.fastlege.FastlegeClient
+import no.nav.syfo.behandler.fastlege.FastlegeResponse
+import no.nav.syfo.behandler.fastlege.toFastlege
+import no.nav.syfo.behandler.partnerinfo.PartnerinfoClient
+import no.nav.syfo.behandler.partnerinfo.PartnerinfoResponse
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.testhelper.*
+import no.nav.syfo.testhelper.generator.generateFastlegeResponse
+import no.nav.syfo.testhelper.generator.generatePartnerinfoResponse
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldNotBeEqualTo
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class BehandlerServiceSpek : Spek({
+
+    val anyToken = "token"
+    val anyCallId = "callId"
+    val partnerId = 1
+    val otherPartnerId = 2
+    val parentHerId = 7
+    val otherParentHerId = 8
+
+    fun FastlegeClient.mockResponse(personIdentNumber: PersonIdentNumber, response: FastlegeResponse) {
+        coEvery {
+            this@mockResponse.fastlege(
+                personIdentNumber,
+                anyToken,
+                anyCallId
+            )
+        } returns response
+    }
+
+    fun PartnerinfoClient.mockResponse(parentHerId: Int, response: PartnerinfoResponse) {
+        coEvery {
+            this@mockResponse.partnerinfo(
+                parentHerId.toString(),
+                anyToken,
+                anyCallId
+            )
+        } returns response
+    }
+
+    describe("BehandlerService") {
+        with(TestApplicationEngine()) {
+            start()
+
+            val fastlegeClientMock = mockk<FastlegeClient>()
+            val partnerinfoClientMock = mockk<PartnerinfoClient>()
```

Comment:
Hadde vært nice med mock på server-nivå her i stedet for mockk, for best mulig testdekning, men kan jo evt. ta det i en senere PR.

Reply:
Hmja. Bruker jo servermocks i API-speken, men hadde lyst på en egen service-test for å dekke flere corner-caser der og da virket det greiest med mockk. Men kan undersøke det i ny PR 👍